### PR TITLE
Setup.py Updates & Boiler-plating

### PR DESCRIPTION
**Pull request by ric-evans**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,48 +1,84 @@
...
+        "License :: OSI Approved :: MIT License",
+        "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.3",
+        "Programming Language :: Python :: 3.4",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
```
Comment (dsschult) [inline setup.py:76]: Should probably adjust these to match PY_VERSION

File: setup.py [hunk 2]
```diff
@@ -1,48 +1,84 @@
...
-        'Development Status :: 4 - Beta',
-        'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
+        "Development Status :: 4 - Beta",
```
Comment (dsschult) [inline setup.py:66]: This could probably be tied to the version number?  I assume anything over 1.0.0 is out of beta.

File: setup.py (hunk 1 above)
Comment (ric-evans) [reply setup.py:76]: do we need all the python version or just the current one?

Comment (ric-evans) [reply setup.py]: I'm including everything from 3.3+. I don't know why we are excluding 3.1-3.2, but I didn't touch that.

Comment (dsschult) [reply setup.py]: I would only include the ones we expect it to work on.  So some of the early 3.x probably don't work anymore.

Comment (dsschult) [reply setup.py]: Specifically, should python2 be removed now, especially if you start putting in f-strings?

Comment (ric-evans) [reply setup.py]: is there a way to test this? Otherwise, I'd just say 3.5

Comment (dsschult) [reply setup.py]: Not easily?  You could probably make a bot try out installation/tests on each python version.  But that sounds painful.  You could also do "anything >= PY_VERSION" as a safe bet on what will work.

Comment (ric-evans) [reply setup.py]: I added a `OLDEST_PY_VERSION` tuple

File: setup.py (hunk 2 above)
Comment (ric-evans) [reply setup.py]: I added cases for 2-5

File: setup.py (hunk 1 above)
Comment (blinkdog) [reply setup.py]: Yeah, Python 3.6 is minimum for f-strings

**Review by dsschult — APPROVED** [review]

**Pull request merged**